src/store/diagnostics.ts

```diff
@@ -14,6 +14,19 @@ interface DiagnosticsState {
 
 const uri = (port: number, pathRemainder: string) => `http://localhost:${ port }/v0/${ pathRemainder }`;
 
+/**
+ * Updates the muted property for diagnostic results.
+ * @param checks A collection of diagnostic results that require muting.
+ * @param mutedChecks A collection of key, value pairs that contains a key of
+ * the ID for the diagnostic and a boolean value for muting the result.
+ * @returns A collection of diagnostic results with an updated muted property.
+ */
+const mapMutedDiagnostics = (checks: DiagnosticsResult[], mutedChecks: Record<string, boolean>) => {
+  return checks.map((check) => {
+    return check.id in mutedChecks ? { ...check, mute: mutedChecks[check.id] } : check;
+  });
+};
+
 export const state: () => DiagnosticsState = () => (
```

Comment:
DiagnosticsState should now be something like:
```ts

interface DiagnosticsState {
  diagnostics: Array<DiagnosticsResult & { mute: boolean }>,
  timeLastRun: Date;
  inError: boolean;
}
```

Reply:
Phil brought up that this makes no sense — `DiagnosticsResult` _already_ has a `mute` property.